src/soundfile.js

```diff
@@ -539,6 +539,8 @@ define(function (require) {
     var pTime = time + now;
 
     if (this.isPlaying() && this.buffer && this.bufferSourceNode) {
+      this._workletNode.port.postMessage({ name: 'pause' });
```

Comment:
in Safari, this message doesn't always update the worker's `paused` attribute before it can post a `position` message of 0. As a result, `SoundFile._lastPos` gets set to 0 and so `SoundFile.currentTime()` returns 0 while the soundfile is paused. It's inconsistent, happens maybe half the time, and only in Safari.

It resumes playback from the proper position because `pauseTime` gets set below. Maybe we also/instead need to set a paused attribute on the SoundFile rather than the Processor?

Reply:
The solution I arrived at was to remove the pause logic from the processor and add a guard statement in the worklet node's port callback that returns early if the SoundFile is paused or the reported playback position is 0, which handles the race condition that you encountered in Safari.

Because the processor always looks for the last position value in the current frame of the counter buffer, the playback position will only ever be 0 if the SoundFile is paused (i.e., on the 1st frame, the playback position will be 128/256/whatever the buffer size of the worklet processor is).

Reply:
https://github.com/processing/p5.js-sound/pull/380/commits/ce48b1b4c12241fc2ce3445331fe9a06c502f872#diff-635fd3ddb6f066be99025f1994db3e30R1251

Reply:
> the playback position will only ever be 0 if the SoundFile is paused

I'm not sure if this is the case in Firefox, where if you pause and un-pause rapidly, the soundfile will stutter at the same position (including position 0), rather than advancing one buffer frame at a time. That's a tiny bug but probably not a blocker on this PR.